private/rules/maven_utils.bzl

```diff
@@ -95,6 +117,10 @@ def generate_pom(
         "{scope}": unpacked_coordinates.scope or "compile",
     }
 
+    for key in exclusions:
+        if key not in versioned_dep_coordinates and key not in unversioned_dep_coordinates:
+            fail("Key %s in exclusions does not occur in versioned_dep_coordinates or unversioned_dep_coordinates" % key)
```

Comment:
This check is redundant with the check in `get_coordinates` in `pom_file.bzl` but `generator_pom` has another call site, so we've kept this check here.